source/extensions/common/wasm/context.cc

```diff
@@ -950,8 +950,12 @@ WasmResult Context::setProperty(absl::string_view path, absl::string_view value)
 
 WasmResult Context::declareProperty(absl::string_view path,
                                     std::unique_ptr<const WasmStatePrototype> state_prototype) {
-  state_prototypes_[path] = std::move(state_prototype);
-  return WasmResult::Ok;
+  // Do not delete existing schema since it can be referenced by state objects.
+  if (state_prototypes_.find(path) == state_prototypes_.end()) {
+    state_prototypes_[path] = std::move(state_prototype);
+    return WasmResult::Ok;
+  }
```

Comment:
It should be bad argument only if the new and the old registration is different, right?

Reply:
That would require equality for the wasm prototypes. In general, it's not clear to me what to do if this method fails in the client. So the client should just log and ignore.

Reply:
I'd like to get rid of protobuf as an argument eventually. The return result is currently ignored.

Reply:
SGTM.